lib/src/inherited_provider.dart

```diff
@@ -123,8 +123,13 @@ class InheritedProvider<T> extends SingleChildStatelessWidget {
 
   @override
   Widget buildWithChild(BuildContext context, Widget child) {
-    assert(child != null,
-        '$runtimeType used outside of MultiProvider must specify a child');
+    assert(() {
+      if (_builder == null && child == null) {
+        throw AssertionError(
+            '$runtimeType used outside of MultiProvider must specify a child');
+      }
+      return true;
+    }());
```

Comment:
Might be just be, but this is easier to understand than

```dart
assert(_builder != null || child != null, 
    '$runtimeType used outside of MultiProvider must specify a child');
```

Reply:
Why not:

```dart
assert(_builder == null && child == null,
        '$runtimeType used outside of MultiProvider must specify a child');
```

?